File: backend/app/schemas/admin_owners.py
"""`GET /admin/owners` — see docs/API_CONTRACTS.md "GET /admin/owners".
Backs the admin console's "Owners" report (`/admin/owners`), the owner-side
counterpart to `admin_registered_users.py`. Purely local-DB data
(`owner_account` + its brands/locations/follows/pending requests) — no
Cognito call — and deliberately carries NO billing/payment fields
(Stripe/`is_paid` work is deferred; nothing here may leak
`stripe_customer_id`/`stripe_sub_id`).
"""
from __future__ import annotations

from datetime import datetime
from typing import Literal

from pydantic import BaseModel

from app.schemas.admin_overview import StatusBreakdown

OwnerSort = Literal["newest", "oldest", "most_locations", "email"]


class AdminOwnerOut(BaseModel):
    id: int
    # `null` for an owner whose CCPA deletion was executed
    # (`personal_data_deleted=true`): the row's stored value is only a
    # synthetic tombstone (`deleted-owner-<id>@deleted.swarasa.invalid`),
    # which is never surfaced.
    email: str | None
    full_name: str | None
    phone: str | None
    # `owner_account.created_at`.
    joined_at: datetime
    # True once a CCPA deletion request for this owner was approved
    # (`owner_account.personal_data_deleted_at IS NOT NULL`). The row is
    # kept (brands/locations hang off it) but identity fields are redacted.
    personal_data_deleted: bool
    brand_count: int
    # Location-grain, same 4-value status lifecycle as
    # `GET /admin/overview`'s owner breakdown; values sum to `location_count`.
    location_count: int
    by_status: StatusBreakdown
    verified_location_count: int
    # Total `user_follow` rows across all of this owner's brands (a diner
    # following two of the owner's brands counts twice). Admin-only stat —
    # same gate as `RestaurantOut.follower_count`.
    follower_count: int
    # `claim_request` rows in `pending_review` submitted by this owner's
    # Cognito identity (claimant_user_id == owner_account.cognito_sub).
    pending_claim_count: int
    # `location_reopen_request` rows in `pending_review` for locations
    # under this owner's brands.
    pending_reopen_request_count: int


class AdminOwnersResponse(BaseModel):
    results: list[AdminOwnerOut]
    page: int
    page_size: int
    total: int

File: frontend/src/types/adminOwners.ts
// Types for GET /admin/owners, matching docs/API_CONTRACTS.md
// "GET /admin/owners" -- the admin "Owners" report. No billing fields by
// design (payments are deferred).
import type { PaginatedResponse } from "@/types/common";

export type OwnerSort = "newest" | "oldest" | "most_locations" | "email";

export interface OwnerStatusBreakdown {
  active: number;
  owner_deactivated: number;
  coming_soon: number;
  closed_pending_reopen: number;
}

export interface AdminOwnerRow {
  id: number;
  /** `null` when `personal_data_deleted` (CCPA deletion executed). */
  email: string | null;
  full_name: string | null;
  phone: string | null;
  /** ISO timestamp -- `owner_account.created_at`. */
  joined_at: string;
  personal_data_deleted: boolean;
  brand_count: number;
  /** Location-grain; `by_status` values sum to this. */
  location_count: number;
  by_status: OwnerStatusBreakdown;
  verified_location_count: number;
  /** Total follows across all of this owner's brands. */
  follower_count: number;
  pending_claim_count: number;
  pending_reopen_request_count: number;
}

export type AdminOwnersResponse = PaginatedResponse<AdminOwnerRow>;

export interface AdminOwnersParams {
  page?: number;
  page_size?: number;
  /** Case-insensitive substring match on email or full name. */
  q?: string;
  sort?: OwnerSort;
}
